### PR TITLE
[SMALLFIX] Fix missing configuration doc and template

### PR DESCRIPTION
**Pull request by apc999**

### CODE REVIEW

File: docs/en/Configuration-Settings.md [hunk 1]
```diff
@@ -63,6 +65,10 @@ default <code class="highlighter-rouge">ALLUXIO_JAVA_OPTS</code> is included in
 <code class="highlighter-rouge">ALLUXIO_MASTER_JAVA_OPTS</code> and
 <code class="highlighter-rouge">ALLUXIO_WORKER_JAVA_OPTS</code>.</td>
 </tr>
+<tr>
+  <td><code class="highlighter-rouge">ALLUXIO_USER_JAVA_OPTS</code></td>
+  <td>additional Java VM options for client configuration.</td>
```
Comment (aaudiber) [inline docs/en/Configuration-Settings.md]: client + shell configuration?

**Comment by aaudiber** [comment]
LGTM module one question

File: docs/en/Configuration-Settings.md (hunk 1 above)
Comment (apc999) [inline docs/en/Configuration-Settings.md]: good catch. this should be shell configuration rather than client

**Comment by apc999** [comment]
@aaudiber Addressed your comments. thanks!

**Comment by apc999** [comment]
@aaudiber PTAL?

File: docs/en/Configuration-Settings.md [hunk 2]
```diff
@@ -13,15 +13,39 @@ for Alluxio in different contexts.
...
 
-Alluxio runtime respects two sources of configuration settings:
+Alluxio runtime respects three sources of configuration settings:
 
-1. [Environment variables](#Environment-variables). This is a convenient way for beginner users or shell scripts to set
-a few basic properties when running Alluxio.
-2. [Configuration properties](#configuration-properties). This provides a general way to customize any
-[supported Alluxio configure properties](#appendix).
+1. [Application settings](#application-settings). Setting Alluxio configuration in this way is application-specific,
+and is required each time when running a new Spark/MapReduce job or an Alluxio shell command.
```
Comment (jsimsa) [inline docs/en/Configuration-Settings.md]: replace `Spark/MapReduce job or an Alluxio shell command` with `application instance`

File: docs/en/Configuration-Settings.md [hunk 3]
```diff
@@ -13,15 +13,39 @@ for Alluxio in different contexts.
...
+2. [Environment variables](#environment-variables). This is an easy and fast way to set the basic properties
+to manage Alluxio servers and run Alluxio shell commandlines.
```
Comment (jsimsa) [inline docs/en/Configuration-Settings.md]: s/commandlines/commands

File: docs/en/Configuration-Settings.md [hunk 4]
```diff
@@ -13,15 +13,39 @@ for Alluxio in different contexts.
...
+
+The priority to load property values, from the highest to the lowest, is
+application settings (if any), environment variables, property files and the defaults.
+
+## Application settings
+
+Alluxio shell users can use `-Dkey=property` to specify an Alluxio configuration value in commandline. For example,
+
+```java
+$ bin/alluxio fs -Dalluxio.user.file.writetype.default=MUST_CACHE touch /foo
```
Comment (jsimsa) [inline docs/en/Configuration-Settings.md]: FYI, it is odd that this key contains "default" as a suffix ... we are using it here to override the default

File: docs/en/Configuration-Settings.md [hunk 5]
```diff
@@ -13,15 +13,39 @@ for Alluxio in different contexts.
...
+Spark users can add `"-Dkey=property"` to `${SPARK_DAEMON_JAVA_OPTS}` in `conf/spark-env.sh`, or add it to
+`spark.executor.extraJavaOptions` (for Spark executors) and `spark.driver.extraJavaOptions` (for Spark drivers).
+
+Hadoop MapReduce users can set `"-Dkey=property"` in `hadoop jar` command-lines to pass it down to Alluxio:
+
+```java
+$ hadoop jar -Dalluxio.user.file.writetype.default=MUST_CACHE foo.jar
+```
+
+Note that, setting Alluxio configuration in this way is application specific and is also required for each job or command.
```
Comment (jsimsa) [inline docs/en/Configuration-Settings.md]: remove `also`

**Comment by jsimsa** [comment]
thanks for these improvements ... I left some minor comments

File: docs/en/Configuration-Settings.md (hunk 4 above)
Comment (apc999) [inline docs/en/Configuration-Settings.md]: agree. we should  probably just call it `alluxio.user.file.writetype`

File: docs/en/Configuration-Settings.md (hunk 2 above)
Comment (apc999) [inline docs/en/Configuration-Settings.md]: done

File: docs/en/Configuration-Settings.md (hunk 3 above)
Comment (apc999) [inline docs/en/Configuration-Settings.md]: done

File: docs/en/Configuration-Settings.md (hunk 5 above)
Comment (apc999) [inline docs/en/Configuration-Settings.md]: done

**Comment by apc999** [comment]
@jsimsa I addressed your comments. PTAL

**Comment by jsimsa** [comment]
LGTM

**Pull request merged**